kart/reflink_util.py

```diff
@@ -0,0 +1,40 @@
+import shutil
+
+import reflink as rl
+
+
+def reflink(from_, to):
+    """Same as reflink.reflink, but can be called with pathlib.Path objects."""
+    rl.reflink(str(from_), str(to))
+
+
+def try_reflink(*args):
+    """
+    Usage:
+    >>> try_reflink(from_, to) - tries to use reflink to copy, but falls back to shutil.copy
+    >>> copy_ = try_reflink() - returns a callable object that tries to use reflink to copy until
+        that fails, and from then on it uses shutil.copy
+    """
+
+    reflink_copier = ReflinkCopier()
+    if args:
+        return reflink_copier(*args)
+    else:
+        return reflink_copier
+
+
+class ReflinkCopier:
+    """
+    A callable object that, when called with (from_, to), calls reflink(from_, to) -
```

Comment:
Is this necessary? I can see the usefulness _if_ trying and failing to create a reflink is expensive, otherwise I'd say we should just try to reflink each file and then copy it if that doesn't work out

Reply:
Done